compiler/haskell-ide-core/src/Development/IDE/State/Shake.hs

```diff
@@ -184,15 +197,17 @@ setValues :: IdeRule k v
           -> IO (Maybe [Diagnostic], [Diagnostic]) -- ^ (before, after)
 setValues state key file val = modifyVar state $ \inVal -> do
     let k = Key key
-        outVal = Map.insertWith Map.union file (Map.singleton k $ fmap toDyn <$> val) inVal
+        outVal = Map.insertWith Map.union file (Map.singleton k $ second (fmap toDyn) val) inVal
         f = concatMap fst . Map.elems
     return (outVal, (f <$> Map.lookup file inVal, f $ outVal Map.! file))
 
-getValues :: forall k v. IdeRule k v => Var Values -> k -> FilePath -> IO (Maybe (IdeResult v))
-getValues state key file = flip fmap (readVar state) $ \vs -> do
-    f <- Map.lookup file vs
-    k <- Map.lookup (Key key) f
-    pure $ fmap (fromJust . fromDynamic) <$> k
+getValues :: forall k v. IdeRule k v => Var Values -> k -> FilePath -> IO (Maybe (Maybe v))
```

Comment:
What does the Maybe (Maybe v) mean? For example what does Just Nothing tell me?

Reply:
Just Nothing tells you that it has been computed, but that last time it computed, it failed to produce a value. There's no point rerunning it because it has working dependencies, so it will fail again. In contrast, Nothing means it hasn't ever been computed.